src/Sowp/NewsModuleBundle/Tests/Controller/NewsControllerTest.php

```diff
@@ -0,0 +1,153 @@
+    }
+
+    /**
+     *  response and page header.
+     */
+    public function testIndexHeader()
+    {
+        echo __FUNCTION__."\n";
+        $client = $this->createClient();
+        $crawler = $client->request('GET', '/wiadomosci/');
+
+        $this->assertEquals(200, $client->getResponse()->getStatusCode());
+        $this->assertGreaterThan(
+            0,
+            $crawler->filter('html:contains("News list")')->count(),
+            'HTML doesn\'t contain proper heading'
+        );
+    }
+
+    /**
+     * structure of news index.
+     */
+    public function testIndexMessageStructure()
+    {
+        echo __FUNCTION__."\n";
+        $client = $this->createClient();
+        $crawler = $client->request('GET', '/wiadomosci/');
+        $count = (int) $this->news_R->getTotalNewsCount();
+        $divs = $crawler->filter('div.news-entry');
+        $divs_c = $divs->count();
+
+        if ($count > 0) {
+            $this->assertGreaterThan(0, $divs->count(), 'Count of News objects is greater than 0,
+                                                         but there are no corresponding DOM elements');
+
+            foreach ($divs as $child) {
+                /* @var $child \DOMElement */
+                $this->assertEquals(
+                    1, $child->getElementsByTagName('table')->length,
+                    "No '<table>' element found in iteration throug div.art-entry"
+                );
+            }
+
+            $id_c = $divs->filter('td.id')->count();
+            $title_c = $divs->filter('h3.panel-title')->count();
+
+            $this->assertEquals($divs_c, $id_c, 'Inapropriate structure - lacking td.id');
+            $this->assertEquals($divs_c, $title_c, 'Inapropriate structure - lacking h3.panel-title');
+        } else {
+            $this->assertEquals(0, $divs->count());
+        }
+    }
+
+    public function testAddNews()
+    {
+        echo __FUNCTION__."\n";
+        $client = $this->createClient();
+        $client->followRedirects();
+        $crawler = $client->request('GET', '/wiadomosci/dodaj');
+        $form = $crawler->selectButton('send_new_message')->form();
+        $coll_ids = [];
+        $faker = \Faker\Factory::create();
+
+        foreach ($this->cat_R->getCollectionsIds() as $id) {
+            $coll_ids[] = $id['id'];
+        }
+
+        $this->assertEquals(200, $client->getResponse()->getStatusCode(),
+                            "Route '/wiadomosci/dodaj' should return HTTP 200");
+
+        foreach ($form->all() as $input) {
+            $field_name = $input->getName();
+            switch ($field_name) {
+                case \preg_match('#title#', $field_name) === 1:
+                    $form[$field_name] = 'Testing title at '.\time();
+                    break;
+                case preg_match('#content#', $field_name) === 1:
+                    $str = '';
+                    $x1 = \rand(10, 500);
+                    for ($x = $x1; $x >= 0; --$x) {
```

Comment:
Generuje od 10 do 500 lini o długości zadanej poniżej,
To jest jakaś różnica czy pętla jest od 0 do 500 czy od 500 do 0, tak poprostu m i się napisało?
Nie wiem co jest generowane dwa razy????
Duża pętla idzie przez pola formularza wyszukując odpowiednie -
ta wiadomo napisane powyżej.